### ISSUE TITLE
Consumer acl POST or PUT with an existing group should return 409 (not 400)

### ISSUE THREAD

**Issue by dinesh-hobsons** [Kong/kong#3818]
NOTE: GitHub issues are reserved for **bug reports only**. For anything else,
please join the conversation in Kong Nation https://discuss.konghq.com/.

Please read the CONTRIBUTING.md guidelines to learn on which channels you can
seek for help and ask general questions:

https://github.com/Kong/kong/blob/master/CONTRIBUTING.md#where-to-seek-for-help

### Summary

Consumer acl POST or PUT with an existing group should return 409 (not 400)

### Steps To Reproduce

1. curl -v http://localhost:8001/consumers/{consumer}/acls --data "group=group1"
2. Run the same command
curl -v http://localhost:8001/consumers/{consumer}/acls --data "group=group1"
3. Return 400 (Bad Request) 
upload completely sent off: 12 out of 12 bytes
< HTTP/1.1 400 Bad Request
< Content-Type: application/json; charset=utf-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< Access-Control-Allow-Origin: *
< Server: kong/0.12.3
< 
{"group":"ACL group already exist for this consumer"}

4. It should probably return 409 to be consistent with other APIs

**Comment by bungle** [comment]
The whole situation with `PUT` has changed a lot since `0.12.3`. It was never properly implemented on `0.12.3`. Starting with a fct that `PUT` should not be a `collection` endpoint, but on `entity` endpoint (if you are not going to replace the whole collection, which we don't support).

This is what you get on current version:

```bash
$ http put :8001/consumers/demo
HTTP/1.1 200 OK
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 105
Content-Type: application/json; charset=utf-8
Date: Fri, 05 Oct 2018 10:46:05 GMT
Server: kong/0.14.1

{
    "created_at": 1538736365,
    "custom_id": null,
    "id": "bc2d631a-89b6-4a80-8f5f-c4d9da5da46c",
    "username": "demo"
}

$ http post :8001/consumers/demo/acls group=group1 -f
HTTP/1.1 201 Created
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 144
Content-Type: application/json; charset=utf-8
Date: Fri, 05 Oct 2018 10:46:31 GMT
Server: kong/0.14.1

{
    "consumer": {
        "id": "bc2d631a-89b6-4a80-8f5f-c4d9da5da46c"
    },
    "created_at": 1538736391,
    "group": "group1",
    "id": "daf7292a-160e-47c0-a3bf-304dd2433101"
}

$ http post :8001/consumers/demo/acls group=group1 -f
HTTP/1.1 409 Conflict
Access-Control-Allow-Origin: *
Connection: keep-alive
Content-Length: 250
Content-Type: application/json; charset=utf-8
Date: Fri, 05 Oct 2018 10:46:33 GMT
Server: kong/0.14.1

{
    "code": 5,
    "fields": {
        "consumer": {
            "id": "bc2d631a-89b6-4a80-8f5f-c4d9da5da46c"
        },
        "group": "group1"
    },
    "message": "UNIQUE violation detected on '{consumer={id=\"bc2d631a-89b6-4a80-8f5f-c4d9da5da46c\"},group=\"group1\"}'",
    "name": "unique constraint violation"
}
```

**Comment by bungle** [comment]
I consider this to be fixed as `1.0.0` is now released. @dinesh-hobsons, please let us know if you still think there are issues with this. Thank you.
